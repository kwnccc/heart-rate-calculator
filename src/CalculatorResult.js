//@flow

import React from 'react';

import List from 'react-toolbox/lib/list/List';
import ListItem from 'react-toolbox/lib/list/ListItem';
import ListSubHeader from 'react-toolbox/lib/list/ListSubHeader';
import ListDivider from 'react-toolbox/lib/list/ListDivider';

import type { CalculatorResultType } from './CalculatorTypes';

const CalculatorResult = ({mhr=0, rrh=0, zones=[]} : CalculatorResultType) => (
  <List ripple={false}>
    <ListSubHeader caption="Results" />
    <ListItem ripple={false} caption='Max Heart Rate' legend={`${mhr}`}/>
    <ListItem ripple={false} caption='Heart Rate Reserve' legend={`${rrh}`}/>
    <ListDivider />
    {zones.map((zone, i) => {
        return <ListItem key={i} ripple={false} caption={`Zone ${i+1}`} legend={`From ${zone.min} to ${zone.max}`} />
    })}
  </List>
);

export default CalculatorResult;
