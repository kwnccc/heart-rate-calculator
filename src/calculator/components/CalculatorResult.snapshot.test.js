import React from 'react';
import renderer from 'react-test-renderer';
import CalculatorResult from './CalculatorResult';

test('CalculatorResult renders with default values', () => {
  const tree = renderer.create(
    <CalculatorResult />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('CalculatorResult renders with prop values', () => {
  const tree = renderer.create(
    <CalculatorResult mhr={195} rrh={137} zones={[{min:105,max:122},{min:123,max:134}]}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
