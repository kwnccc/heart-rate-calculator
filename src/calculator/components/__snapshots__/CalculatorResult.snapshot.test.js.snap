exports[`test CalculatorResult renders with default values 1`] = `
<ul
  className=""
  data-react-toolbox="list">
  <h5
    className="">
    Results
  </h5>
  <li
    className="undefined "
    onClick={[Function]}
    onMouseDown={[Function]}
    onTouchStart={[Function]}>
    <span
      className="">
      <span
        className="">
        <span
          className="undefined"
          data-react-toolbox="list-item-text">
          Max Heart Rate
        </span>
        <span
          className=""
          data-react-toolbox="list-item-text">
          0
        </span>
      </span>
    </span>
  </li>
  <li
    className="undefined "
    onClick={[Function]}
    onMouseDown={[Function]}
    onTouchStart={[Function]}>
    <span
      className="">
      <span
        className="">
        <span
          className="undefined"
          data-react-toolbox="list-item-text">
          Heart Rate Reserve
        </span>
        <span
          className=""
          data-react-toolbox="list-item-text">
          0
        </span>
      </span>
    </span>
  </li>
  <hr
    className={undefined} />
</ul>
`;

exports[`test CalculatorResult renders with prop values 1`] = `
<ul
  className=""
  data-react-toolbox="list">
  <h5
    className="">
    Results
  </h5>
  <li
    className="undefined "
    onClick={[Function]}
    onMouseDown={[Function]}
    onTouchStart={[Function]}>
    <span
      className="">
      <span
        className="">
        <span
          className="undefined"
          data-react-toolbox="list-item-text">
          Max Heart Rate
        </span>
        <span
          className=""
          data-react-toolbox="list-item-text">
          195
        </span>
      </span>
    </span>
  </li>
  <li
    className="undefined "
    onClick={[Function]}
    onMouseDown={[Function]}
    onTouchStart={[Function]}>
    <span
      className="">
      <span
        className="">
        <span
          className="undefined"
          data-react-toolbox="list-item-text">
          Heart Rate Reserve
        </span>
        <span
          className=""
          data-react-toolbox="list-item-text">
          137
        </span>
      </span>
    </span>
  </li>
  <hr
    className={undefined} />
  <li
    className="undefined "
    onClick={[Function]}
    onMouseDown={[Function]}
    onTouchStart={[Function]}>
    <span
      className="">
      <span
        className="">
        <span
          className="undefined"
          data-react-toolbox="list-item-text">
          Zone 1
        </span>
        <span
          className=""
          data-react-toolbox="list-item-text">
          From 105 to 122
        </span>
      </span>
    </span>
  </li>
  <li
    className="undefined "
    onClick={[Function]}
    onMouseDown={[Function]}
    onTouchStart={[Function]}>
    <span
      className="">
      <span
        className="">
        <span
          className="undefined"
          data-react-toolbox="list-item-text">
          Zone 2
        </span>
        <span
          className=""
          data-react-toolbox="list-item-text">
          From 123 to 134
        </span>
      </span>
    </span>
  </li>
</ul>
`;
